docs/reference/pip_install.rst

```diff
@@ -445,19 +445,100 @@ When no wheels are found for an sdist, pip will attempt to build a wheel
+  strength, for legacy reasons), though you should use a stronger
+  hash like sha256 whenever possible.
+* Hashes are required for all dependencies. An error is raised if there is a
+  dependency that is not spelled out and hashed in the requirements file.
+* Requirements that take the form of project names (rather than URLs or local
+  filesystem paths) must be pinned to a specific version using ``==``. This
+  prevents a surprising hash mismatch upon the release of a new version
+  that matches the requirement specifier.
+* ``--egg`` is disallowed, because it delegates installation of dependencies
+  to setuptools, giving up pip's ability to enforce any of the above.
+
+Hash-checking mode can be forced on with the ``--require-hashes`` command-line
+option::
+
+    $ pip install --require-hashes -r requirements.txt
+        ...
+        Hashes are required in --require-hashes mode (implicitly on when a hash is
+        specified for any package). These requirements were missing hashes,
+        leaving them open to tampering. These are the hashes the downloaded
+        archives actually had. You can add lines like these to your requirements
+        files to prevent tampering.
+            pyelasticsearch==1.0 --hash=sha256:44ddfb1225054d7d6b1d02e9338e7d4809be94edbe9929a2ec0807d38df993fa
+            more-itertools==2.2 --hash=sha256:93e62e05c7ad3da1a233def6731e8285156701e3419a5fe279017c429ec67ce0
+
+This can be useful in deploy scripts, to ensure that the author of the
+requirements file provided hashes. It is also a convenient way to bootstrap
+your list of hashes, since it shows the hashes of the packages it fetched. It
+fetches only the preferred archive for each package, so you may still need to
+add hashes for alternatives archives using :ref:`pip hash`: for instance if
+there is both a binary and a source distribution.
+
+Hash-checking mode also functions with :ref:`pip download` and :ref:`pip
+wheel`. A :ref:`comparison of hash-checking mode with other repeatability
+strategies <Repeatability>` is available in the User Guide.
+
+.. warning::
+    Beware of the ``setup_requires`` keyword arg in :file:`setup.py`. The
+    (rare) packages that use it will cause those dependencies to be downloaded
+    by setuptools directly, skipping pip's hash-checking. If you need to use
+    such a package, see :ref:`Controlling
+    setup_requires<controlling-setup-requires>`.
+
+.. warning::
+    Be careful not to nullify all your security work when you install your
+    actual project. If you call ``python setup.py install`` after installing
+    your requirements, setuptools will happily go out and download, unchecked,
+    anything you missed in your requirements file—and it’s easy to miss things
+    as your project evolves. One way to be safe is to pack up your project and
+    then install that using pip and :ref:`--no-deps <install_--no-deps>`::
+
+        python setup.py sdist
+        pip install --no-deps dist/yourproject-1.0.tar.gz
```

Comment:
btw, you don't need to build and install separately like this (see below).  it's fine, but you don't have to. 

you could say something like this.

"Instead of using `python setup.py develop`, you should use::

    pip install --no-deps -e .

Instead of using  `python setup.py install`, you should use::

    pip install --no-deps .

or::
     python setup.py sdist
     pip install --no-deps dist/yourproject-1.0.tar.gz